src/srpo_uci.c

```diff
@@ -68,104 +110,107 @@ const char *srpo_uci_error_description_get(srpo_uci_error_e error)
+	for (int i = 0; i < node_sec->ch_nr; i++) {
+		uci2_n_t *child = node_sec->ch[i];
+		if (child->nt == UCI2_NT_LIST) {
+			// if its not an option -> list
+			size_t byte_cnt = 0;
 
-					if (anonymous_section_exists == false) { // add the anonymous section to the list if first occurrence
-						anonymous_section_list = xrealloc(anonymous_section_list, (anonymous_section_list_size + 1) * sizeof(struct anonymous_section));
-						anonymous_section_list[anonymous_section_list_size].type = strdup(section->type);
-						anonymous_section_list[anonymous_section_list_size].index = 0;
-						anonymous_section_index = anonymous_section_list[anonymous_section_list_size].index++;
-						anonymous_section_list_size++;
-					}
+			for (int j = 0; j < child->ch_nr; j++) {
+				uci2_n_t *li = child->ch[j];
 
-					section_name_size = strlen(section->type) + 1 + 2 + 20 + 1;
-					section_name = xmalloc(section_name_size);
-					snprintf(section_name, section_name_size, "@%s[%zu]", section->type, anonymous_section_index);
+				if (sizeof(list_items_buffer) - byte_cnt > 0) {
+					snprintf(list_items_buffer + byte_cnt, sizeof(list_items_buffer) - byte_cnt, "\'%s\' ", uci2_get_name(li));
+					byte_cnt += strlen(li->name);
 				} else {
-					section_name = xstrdup(section->e.name);
+					error = SRPO_UCI_ERR_UCI;
+					break;
 				}
+			}
+			// remove last space
+			list_items_buffer[byte_cnt - 2] = 0;
+		}
+		snprintf(path_buffer, sizeof(path_buffer), "%s.%s.%s=%s", uci_config, sec_buffer, uci2_get_name(child), child->nt == UCI2_NT_LIST ? list_items_buffer : uci2_get_value(child));
+		srpo_path_list_append(path_list, xstrdup(path_buffer));
+	}
+	return error;
+}
 
-				uci_foreach_element(&section->options, element_option)
-				{
-					option = uci_to_option(element_option);
+int srpo_uci_ucipath_list_get(const char *uci_config, const char **uci_section_list, size_t uci_section_list_size, char ***ucipath_list, size_t *ucipath_list_size, bool convert_to_extended)
+{
+	int error = 0;
+	srpo_path_list_t path_list;
 
-					uci_path_list_tmp = xrealloc(uci_path_list_tmp, (uci_path_list_size_tmp + 1) * sizeof(char *));
+	srpo_path_list_init(&path_list);
+	error = uci_context_load(uci_context, uci_config);
 
-					uci_path_size = strlen(uci_config) + 1 + strlen(section_name) + 1 + strlen(option->e.name) + 1;
-					uci_path_list_tmp[uci_path_list_size_tmp] = xmalloc(uci_path_size);
-					snprintf(uci_path_list_tmp[uci_path_list_size_tmp], uci_path_size, "%s.%s.%s", uci_config, section_name, option->e.name);
+	if (error != SRPO_UCI_ERR_OK) {
+		return error;
+	} else {
+		uci2_n_t *root = UCI2_CFG_ROOT(uci_context->parser_ctx);
 
-					uci_path_list_size_tmp++;
-				}
+		for (size_t iter = 0; iter < uci_section_list_size; iter++) {
+			for (int i = 0; i < root->ch_nr; i++) {
+				uci2_n_t *type = root->ch[i];
 
-				FREE_SAFE(section_name);
+				// check if types match
+				if (strcmp(type->name, uci_section_list[i]) == 0) {
```

Comment:
There seems to be an out of bounds exception here when `uci_section_list_size < root->ch_nr`. Perhaps `uci_section_list[iter]` is supposed to be used?

Reply:
Fixed in the latest commit. Thanks!

Reply:
Thanks!